instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/__init__.py

```diff
@@ -147,18 +149,33 @@ def _traced_execute_command(func, instance, args, kwargs):
             name = args[0]
         else:
             name = instance.connection_pool.connection_kwargs.get("db", 0)
-        with tracer.start_as_current_span(
-            name, kind=trace.SpanKind.CLIENT
-        ) as span:
-            if span.is_recording():
-                span.set_attribute(SpanAttributes.DB_STATEMENT, query)
-                _set_connection_attributes(span, instance)
-                span.set_attribute("db.redis.args_length", len(args))
-            if callable(request_hook):
-                request_hook(span, instance, args, kwargs)
-            response = func(*args, **kwargs)
-            if callable(response_hook):
-                response_hook(span, instance, response)
+
+        try:
+            with tracer.start_as_current_span(
+                name, kind=trace.SpanKind.CLIENT
+            ) as span:
+                if span.is_recording():
+                    span.set_attribute(SpanAttributes.DB_STATEMENT, query)
+                    _set_connection_attributes(span, instance)
+                    span.set_attribute("db.redis.args_length", len(args))
+                if callable(request_hook):
+                    request_hook(span, instance, args, kwargs)
+                response = func(*args, **kwargs)
+                connections_usage.add(
+                    1,
+                    {
+                        "db.client.connection.usage.state": "used",
+                        "db.client.connection.usage.name": instance.connection_pool.pid,
+                    })
+                if callable(response_hook):
+                    response_hook(span, instance, response)
+        finally:
+            connections_usage.add(
+                -1,
+                {
+                    "db.client.connection.usage.state": "idle",
+                    "db.client.connection.usage.name": instance.connection_pool.pid,
+                })
```

Comment:
Can you decide the state of connection yourself based on the command execution? This can't be correct right?